File: backend/internal/types/user_settings.go
package types

type UserSettings struct {
	ID                 int     `json:"id"`
	UserAccountID      *int    `json:"userAccountID,omitempty"`
	FontSize           float64 `json:"fontSize"`
	IsDarkMode         bool    `json:"isDarkMode"`
	LanguageID         *int    `json:"languageID,omitempty"`
	IsUnlisted         bool    `json:"isUnlisted"`
	IsPullFromEveryone bool    `json:"isPullFromEveryone"`
	PullUserListID     *int    `json:"pullUserListID,omitempty"`
	IsNotifyEveryone   bool    `json:"isNotifyEveryone"`
	NotifyUserListID   *int    `json:"notifyUserListID,omitempty"`
	IsQuickAccess      bool    `json:"isQuickAccess"`
}

File: app/lib/hive/user_settings/user_settings.dart
import 'package:hive/hive.dart';

part 'user_settings.g.dart';

@HiveType(typeId: 1)
class UserSettings extends HiveObject {
  @HiveField(0)
  int id;

  @HiveField(1)
  int userAccountID;

  @HiveField(2)
  double fontSize;

  @HiveField(3)
  bool isDarkMode;

  @HiveField(4)
  int languageID;

  @HiveField(5)
  bool isUnlisted;

  @HiveField(6)
  bool isPullFromEveryone;

  @HiveField(7)
  int? pullUserListID;

  @HiveField(8)
  bool isNotifyEveryone;

  @HiveField(9)
  int? notifyUserListID;

  @HiveField(10)
  bool isQuickAccess;

  UserSettings({
    required this.id,
    required this.userAccountID,
    required this.fontSize,
    required this.isDarkMode,
    required this.languageID,
    required this.isUnlisted,
    required this.isPullFromEveryone,
    this.pullUserListID,
    required this.isNotifyEveryone,
    this.notifyUserListID,
    required this.isQuickAccess,
  });
}
